tests/repository_data/client/test_repository1/metadata/current/targets.json

```diff
@@ -28,7 +28,7 @@
      ],
      "name": "role1",
      "paths": [
-      "file3.txt"
+      "/file3.txt"
```

Comment:
I thought we removed adding a leading path separator to targets paths?

Reply:
Good eye, this was caused by changing:
`repository.targets.delegate('role1', [delegation_public], [target3_filepath])`
to
`repository.targets.delegate('role1', [delegation_public], [], list_of_targets=[os.path.basename(target3_filepath)])`

I made that change because I was trying to do away with the incorrect warning from `generate.py` that:
> 'file3.txt' is not located in the repository's targets directory: '/home/joshuagl/Projects/tuf/tests/repository_data/repository/targets'

I misunderstood the documentation for `delegate()` on first read, therefore I think it makes sense to undo the change to the `delegate()` call in the generate tool and regenerate the repository_data for this PR.

Then separate from this PR I can create a new Issue to clarify the docs and remove the incorrect warning in `repository_tool.targets.delegate()`

Reply:
In fact, I think the warning isn't even incorrect. At the time that we call `delegate()` in `generate.py` the path being delegated is not yet listed in the targets metadata because we are delegating the path before we have added a target file (with `add_target()`) at that path.

This was all misunderstanding on my part, thanks for the diligent review @trishankatdatadog